modules/renter/contractor/contractor_test.go

```diff
@@ -0,0 +1,195 @@
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// newStub is used to test the New function. It implements all of the contractor's
+// dependencies.
+type newStub struct{}
+
+// consensus set stubs
+func (newStub) ConsensusSetPersistentSubscribe(modules.ConsensusSetSubscriber, modules.ConsensusChangeID) error {
+	return nil
+}
+
+// wallet stubs
+func (newStub) NextAddress() (uc types.UnlockConditions, err error) { return }
+func (newStub) StartTransaction() modules.TransactionBuilder        { return nil }
+
+// transaction pool stubs
+func (newStub) AcceptTransactionSet([]types.Transaction) error { return nil }
+
+// hdb stubs
+func (newStub) Host(modules.NetAddress) (settings modules.HostSettings, ok bool) { return }
+func (newStub) RandomHosts(int, []modules.NetAddress) []modules.HostSettings     { return nil }
+
+// TestNew tests the New function.
+func TestNew(t *testing.T) {
+	// Using a stub implementation of the dependencies is fine, as long as its
+	// non-nil.
+	var stub newStub
+	dir := build.TempDir("contractor", "TestNew")
+
+	// Sane values.
+	_, err := New(stub, stub, stub, stub, dir)
+	if err != nil {
+		t.Fatalf("expected nil, got %v", err)
+	}
+
+	// Nil consensus set.
+	_, err = New(nil, stub, stub, stub, dir)
+	if err != errNilCS {
+		t.Fatalf("expected %v, got %v", errNilCS, err)
+	}
+
+	// Nil wallet.
+	_, err = New(stub, nil, stub, stub, dir)
+	if err != errNilWallet {
+		t.Fatalf("expected %v, got %v", errNilWallet, err)
+	}
+
+	// Nil transaction pool.
+	_, err = New(stub, stub, nil, stub, dir)
+	if err != errNilTpool {
+		t.Fatalf("expected %v, got %v", errNilTpool, err)
+	}
+
+	// Bad persistDir.
+	_, err = New(stub, stub, stub, stub, "")
+	if !os.IsNotExist(err) {
+		t.Fatalf("expected invalid directory, got %v", err)
+	}
+
+	// Corrupted persist file.
+	ioutil.WriteFile(filepath.Join(dir, "contractor.json"), []byte{1, 2, 3}, 0666)
+	_, err = New(stub, stub, stub, stub, dir)
+	if _, ok := err.(*json.SyntaxError); !ok {
+		t.Fatalf("expected invalid json, got %v", err)
+	}
+
+	// Corrupted logfile.
+	os.RemoveAll(filepath.Join(dir, "contractor.log"))
+	f, err := os.OpenFile(filepath.Join(dir, "contractor.log"), os.O_CREATE, 0000)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+	_, err = New(stub, stub, stub, stub, dir)
+	if !os.IsPermission(err) {
+		t.Fatalf("expected permissions error, got %v", err)
+	}
+}
+
+// TestContracts tests the Contracts method.
+func TestContracts(t *testing.T) {
+	c := &Contractor{
+		contracts: map[types.FileContractID]Contract{
+			{1}: Contract{ID: types.FileContractID{1}, IP: "foo"},
+			{2}: Contract{ID: types.FileContractID{2}, IP: "bar"},
+			{3}: Contract{ID: types.FileContractID{3}, IP: "baz"},
+		},
+	}
+	for _, contract := range c.Contracts() {
+		if exp := c.contracts[contract.ID]; exp.IP != contract.IP {
+			t.Errorf("contract does not match: expected %v, got %v", exp.IP, contract.IP)
+		}
+	}
+}
```

Comment:
standard concerns of testing the implementation and not the behavior. I think it's best to explore a function like `Contracts()` in a smoke test, where you're creating a contractor and then trying out all of the behaviors together (with production dependencies). Then you know you were creating contracts the natural way, instead of hardcoding them in where they belonged.